crates/partition-store/src/invocation_status_table/mod.rs

```diff
@@ -201,6 +118,36 @@ fn read_invoked_full_invocation_id(
     }
 }
 
+pub(crate) async fn run_invocation_status_v1_migration(storage: &mut PartitionStore) -> Result<()> {
+    let partition_key_range = storage.partition_key_range().clone();
+
+    let mut iterator = storage
+        .run_iterator(
+            "invocation-status-v1-migration",
+            Priority::High,
+            FullScanPartitionKeyRange::<InvocationStatusKeyV1>(partition_key_range),
+            |(mut old_key, mut old_value)| {
+                Ok((
+                    InvocationStatusKeyV1::deserialize_from(&mut old_key)?,
+                    InvocationStatusV1::decode(&mut old_value)?,
+                ))
+            },
+        )
+        .map_err(|_| StorageError::OperationalError)?;
+
+    while let Some(res) = iterator.next().await {
+        let (key, value) = res?;
+        put_invocation_status(
+            storage,
+            &InvocationId::from_parts(*key.partition_key_ok_or()?, *key.invocation_uuid_ok_or()?),
+            &value.0,
+        )?;
+        storage.delete_key(&key)?;
+    }
```

Comment:
An argument could be made for filling up the write batch while iterating instead of preloading everything in memory before creating the writes.

Reply:
@slinkydeveloper can you help me understand this a bit better - the current implementation doesn't buffer much in memory, as I see it it's streaming the results from the async scan and performing one write/one delete at a time? The possible optimization here is to create a write batch to buffer the put/delete ops, and commit it periodically.